includes/blocks/class-sensei-lesson-properties-block.php

```diff
@@ -43,7 +43,22 @@ public function render( array $attributes, string $content ) : string {
 			return $content;
 		}
 
-		$content = '<div class="wp-block-sensei-lms-lesson-properties">';
+		$class_name             = 'wp-block-sensei-lms-lesson-properties';
+		$lm_template_class_name = 'wp-block-sensei-lms-lesson-properties--lm-template-part';
+		$final_class_name       = $class_name;
+		$style                  = '';
+		if ( true === $attributes['lmTemplatePart'] ) {
+			$final_class_name .= " {$lm_template_class_name}";
+			$style             = "
+				<style>
+					.{$class_name}:not(.{$lm_template_class_name}) {
+						display: none;
+					}
+				</style>
+			";
+		}
```

Comment:
I don't think that we should bother hiding the block. The user might want to have it in many places for some of the lessons. They might also want to simply try it in the lesson content to see how it looks.

Reply:
Hmm, good point. For some reason I thought the Lesson Properties block is auto inserted into the lesson content. Thus we decided to hide it if there is another Lesson Properties block in the LM template.

The hiding the Lesson Properties block in lesson content when it is present LM template is one of the conditions of the [issue](https://github.com/Automattic/sensei/issues/5666) this PR is addressing.

Pinging @yscik because he is the author of the original [issue](https://github.com/Automattic/sensei/issues/5666). I can't remember why exactly we decided to hide the Lesson Properties from the content when it is present in LM template.

Reply:
Right, I also thought it's added by default. I think it was, but maybe got removed when the lesson setup wizard was added with the pattern options. So it might still be present in some existing lessons, but I'm okay with not bothering with hiding it then.

Reply:
I had to create a separate block for Learning Mode templates. Reusing the same `sensei-lms/lesson-properties` doesn't work because we're registering LM blocks in lesson editors too, which tries to register this block twice and it stops working.

Created the `sensei-lms/learning-mode-lesson-properties` block. It reuses the `render` method of the `Sensei_Lesson_Properties_Block` class.

I removed the hiding lesson content block logic.

Tbh, this block does not make much sense. If you don't set Lesson Properties block in the lesson content, then the Lesson Properties of the template doesn't show up either. Because it depends on the `lesson_complexity` and `lesson_length` values, that could be set only via Lesson Properties block in the Lesson editor.

Reply:
Hmm it seems to me that our only viable option is to have lesson properties as settings in the lesson. Then both blocks could get the values from there. Not sure if it is too much worth though. Any other ideas?

Reply:
Not too much work. It actually works like the way suggest right now. The only difference is that it is a setting for the block instead of the Lesson. So if user wants to set those values they simply need to add the Lesson Properties block to the lesson. Then both Lesson Properties and LM Lesson Properties read those same values.

So, it's not available right there in the Lesson settings but it's available via the block. It's easy enough to add these into the Lesson settings but I am not entirely sure if we should keep adding stuff into the Lesson settings. I think the block based approach is more flexible. The user can add those properties if they want via the block. But the downside is it's not available right away.

My preference would be to keep these properties available via block. Because it's probably not that commonly used, so it probably should not be part of all the lessons by default.

Reply:
More context here: p1664435312057249-slack-C013QUH20TS

In conclusion, we will update the look and description Lesson Properties block in LM template so it aligns with all other blocks in LM template. Instead of showing a placeholder `Length: 10 minutes | Difficulty: None` it will show `Lesson Properties` and the description will say that this block displays lesson properties.

Tbh, after very long discussion the conclusion now sounds kinda obvious. 😄

Reply:
Made some changes:
- Lesson Properties block does not render a content in Learning Mode.
- Learning Mode Lesson Properties block is updated to show `Lesson Properties` text instead of the placeholder `"Length: 10 minutes | Difficulty: None"`.


https://user-images.githubusercontent.com/2578542/193044550-424b3de2-4e49-43ee-b35f-18c04f8d837f.mp4

Reply:
There is still one thing that's left for Lesson Properties block.

When user removes the block from the lesson. The meta values `_lesson_length` & `_lesson_complexity` that's set via Lesson Properties block do not get removed.

So if user adds this block then decides they don't want to show these for this lesson, there is no way to remove it, and it will still show up if the template has Lesson Properties block.

I tried to find a proper way to remove those values when the block is removed from the content. So far could not come up with a solution on the gutenberg side.

We could add a hook on the php side. Each time a lesson post is saved we parse the content, check if the Lesson Properties block is present or not. If not then remove the meta data `_lesson_length` & `_lesson_complexity` for that lesson.

But I would say it's out of the scope of this PR or current project for that matter. I created a task in the Backlog of our project board.